src/db.rs

```diff
@@ -156,7 +146,7 @@ impl DbInner {
 
                 if let Some(entry) = iter.next_entry().await? {
```

Comment:
this makes me wonder whether we could implement filtering directly in the underlying iterator 🤔 it would give us the same guarantees for range scans out of the box